### PR TITLE
Throw event log events

### PR DESCRIPTION
**Pull request by fisx**
Related: #216 (not sure we should close that yet)

Missing:

```haskell
  | EventLogTopicNewPhase         topic Phase Phase
  | EventLogIdeaNewTopic          idea (Maybe topic) (Maybe topic)
```

### CODE REVIEW

**Comment by fisx** [comment]
ready for review and merging!

File: src/Action.hs [hunk 1]
```diff
@@ -343,15 +345,15 @@ deleteUser = update . DeactivateUser
 
 -- * Phase Transitions
 
-topicPhaseChange :: (ActionPhaseChange m) => Topic -> PhaseChange -> m ()
```
Comment (andorp) [inline src/Action.hs]: Please delete the type synonym if not used anymore.

File: src/Action.hs [hunk 2]
```diff
@@ -422,26 +416,56 @@ createTopic proto = do
     eventLogUserCreatesTopic topic
     pure topic
 
+editTopic :: ActionM m => AUID Topic -> EditTopicData -> m ()
+editTopic topicId topic = do
+    update $ EditTopic topicId topic
+    eventLogUserEditsTopic =<< equery (maybe404 =<< findTopic topicId)
```
Comment (andorp) [inline src/Action.hs]: Please use `mquery`.

File: src/Action.hs [hunk 3]
```diff
@@ -422,26 +416,56 @@ createTopic proto = do
...
+
 
 -- * Vote Handling
 
-likeIdea :: ActionAddDb m => AUID Idea -> m ()
-likeIdea ideaId = currentUserAddDb_ (AddLikeToIdea ideaId) ()
+likeIdea :: ActionM m => AUID Idea -> m ()
+likeIdea ideaId = do
+    currentUserAddDb_ (AddLikeToIdea ideaId) ()
+    do idea <- equery $ maybe404 =<< findIdea ideaId
```
Comment (andorp) [inline src/Action.hs]: Please compute use one `equery` as it composes.

File: src/Action.hs [hunk 4]
```diff
@@ -642,42 +665,47 @@ eventLogUserEditsTopic topic = do
...
 eventLogUserVotesOnComment ::
-      (ActionUserHandler m, ActionCurrentTimestamp m, ActionLog m)
-      => Idea -> Comment -> Maybe Comment -> UpDown -> m ()
-eventLogUserVotesOnComment idea comment mcomment v = do
-    uid <- currentUserId
+      (ActionUserHandler m, ActionCurrentTimestamp m, ActionPersist m, ActionLog m)
+      => KeyOf Comment -> UpDown -> m ()
+eventLogUserVotesOnComment ck@(CommentKey _ ideaId parentIds _) v = do
+    uid      <- currentUserId
+    idea     <- equery $ maybe404 =<< findIdea ideaId
```
Comment (andorp) [inline src/Action.hs]: It seems that this part should go under one `equery`.

File: src/Action.hs (hunk 2 above)
Comment (andorp) [inline src/Action.hs]: Please do so in the subsequent occurrences too.

**Comment by fisx** [comment]
I have followed your comments in `likeIdea`, but reject the one on `eventLogUserVotesOnComment`, for the code in the latter is only a work-around until the `EventLogUserVotesOnComment` constructor takes a comment key; see FIXME there.

**Pull request merged**